src/v/storage/disk_log_impl.cc

```diff
@@ -909,18 +909,11 @@ offset_stats disk_log_impl::offsets() const {
     // NOTE: we have to do this because ss::circular_buffer<> does not provide
     // with reverse iterators, so we manually find the iterator
     segment_set::type end;
-    segment_set::type term_start;
     for (int i = (int)_segs.size() - 1; i >= 0; --i) {
         auto& seg = _segs[i];
-        if (!seg->empty()) {
-            if (!end) {
-                end = seg;
-            }
-            // find term start offset
-            if (seg->offsets().term < end->offsets().term) {
-                break;
-            }
-            term_start = seg;
+        if (!seg->empty() && !end) {
```

Comment:
nit: seems like `!end` isn't needed here since we always break once we find `end`

Reply:
Nice catch, will remove `!end`